src/serialization/schema_validator.cpp

```diff
@@ -432,10 +436,12 @@ void schema_validator::validate(const config& cfg, const std::string& name, int
 	// close previous errors and print them to output.
 	print_cache();
 
-	// clear cache
-	auto cache_it = cache_.top().find(&cfg);
-	if(cache_it != cache_.top().end()) {
-		cache_it->second.clear();
+	if (!cache_.empty()) {
+		// clear cache
+		auto cache_it = cache_.top().find(&cfg);
+		if(cache_it != cache_.top().end()) {
+			cache_it->second.clear();
+		}
```

Comment:
These changes are there to support re-using a validator?

Reply:
It stops wesnoth from crashing. When testing addon themes I noticed in some cases this can be called when `cache_` is empty, which causes segfault.